pkg/work/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -208,3 +215,45 @@ func (m *ManifestWorkController) applyAppliedManifestWork(ctx context.Context, w
 	_, err = m.appliedManifestWorkPatcher.PatchSpec(ctx, appliedManifestWork, requiredAppliedWork.Spec, appliedManifestWork.Spec)
 	return appliedManifestWork, err
 }
+
+func (m *ManifestWorkController) manifestWorkCompletion(
+	ctx context.Context, controllerContext factory.SyncContext, manifestWork *workapiv1.ManifestWork,
+) (bool, error) {
+	complete := meta.FindStatusCondition(manifestWork.Status.Conditions, workapiv1.ManifestComplete)
+	if complete != nil && complete.Status == metav1.ConditionTrue {
+		var err error
+		// check if work has TTL set and it has elapsed since completion
+		if ttl, ok := remainingTtl(complete.LastTransitionTime, manifestWork.Spec.DeleteOption); ok {
+			if ttl <= 0 {
+				// Only delete if resourceVersion matches in case complete condition has changed
+				deleteOpts := metav1.DeleteOptions{Preconditions: &metav1.Preconditions{ResourceVersion: &manifestWork.ResourceVersion}}
+				err = m.manifestWorkClient.Delete(ctx, manifestWork.Name, deleteOpts)
```

Comment:
I do not think manifestwork should be deleted on the spoke side. The agent should not have the permission to do that. Would you remove the deletion logic for now. I think if we really need this, it will be a controller on hub that perform the action after ttl in the future.

Reply:
It seems like an arbitrary distinction to say that the spoke can't have delete permission. When it already has full update permissions, so it could for instance remove every manifest from the work. Or add Complete conditions and a TTL to make the hub delete it.

Reply:
We only use it to update finalizer of the work.